src/components/Tables/DomainTable/DomainTable.tsx

```diff
@@ -175,73 +173,67 @@ const DomainTable: React.FC<DomainTableProps> = ({
 
 	// Gets metadata for each NFT in domain list
 	useEffect(() => {
-		const loaded: DomainData[] = [];
-		setHasMetadataLoaded(false);
-		let count = 0;
-		let completed = 0;
+		let loaded: DomainData[] = [];
 
 		// Get metadata
-		const getData = async (domain: Domain) => {
+		const getData = async (domain: Domain[]) => {
 			try {
-				const [metadata, bids] = await Promise.all([
-					getMetadata(domain.metadata),
-					getBidsForDomain(domain),
-				]);
-
-				completed++;
-
-				if (!metadata) {
-					console.log(`found no metadata for ${domain.id}`);
-					return;
-				}
+				for (let i = 0; i < domains.length; i++) {
+					const [metadata, bids] = await Promise.all([
+						getMetadata(domain[i].metadata),
+						getBidsForDomain(domain[i]),
+					]);
+
+					if (!metadata) {
+						console.log(`found no metadata for ${domain[i].id}`);
+						return;
+					}
 
-				// Filter out user's bids if configured to do so
-				let filteredBids;
-				if (hideOwnBids) {
-					filteredBids = bids?.filter(
-						(bid: Bid) => bid.bidderAccount !== userId,
-					);
-				}
+					// Filter out user's bids if configured to do so
+					let filteredBids;
+					if (hideOwnBids) {
+						filteredBids = bids?.filter(
+							(bid: Bid) => bid.bidderAccount !== userId,
+						);
+					}
 
-				loaded.push({
-					domain: domain,
-					metadata: metadata,
-					bids: filteredBids || bids || [],
-				});
+					loaded = loadedDomains;
+					loaded.push({
+						domain: domain[i],
+						metadata: metadata,
+						bids: filteredBids || bids || [],
+					});
 
-				if (completed === count) {
-					setLoadedDomains(loaded);
-					setHasMetadataLoaded(true);
+					setLoadedDomains(loaded); //load on every fetch instead of the last one
+					setIsLoading(true);
 					setIsLoading(false);
+
 					if (onLoad) {
 						onLoad();
 					}
 				}
 			} catch (e) {}
 		};
-
-		for (let i = 0; i < domains.length; i++) {
-			if (!domains[i].metadata) continue;
-			count++;
-			getData(domains[i]);
-		}
-
-		if (!count) {
-			setHasMetadataLoaded(true);
-			setIsLoading(false);
-		}
-	}, [domains]);
+		getData(domains);
+	}, [domainsChanged]); //this was [domains], but that way it triggers a weird bug where the page re render and fetch bad
```

Comment:
This bug is most likely caused due to an outdated hook. You need to make sure that this effect is cleaned up after a previous render if it hasn't completed yet: https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1

This is accomplished using the `isSubscribed` pattern you can find in some other hooks in this project: https://github.com/zer0-os/zNS-dapp/blob/c685f0d795ec31c3d3eb66506aeff53d210f2250/src/components/Profile/Profile.tsx#L86

Essentially, you should never call a `set____` (React set state) function unless you are sure the hook is still valid. This is especially important for long running operations such as this. If for some reason you are no longer subscribed, you just exit out.